packages/core/src/baseclient.ts

```diff
@@ -508,6 +510,7 @@ export abstract class BaseClient<B extends Backend, O extends Options> implement
     // 0.0 === 0% events are sent
     // Sampling for transaction happens somewhere else
     if (!isTransaction && typeof sampleRate === 'number' && Math.random() > sampleRate) {
+      transport.recordLostEvent?.(Outcome.SampleRate);
```

Comment:
s/Lost/Discarded?

Lost might give an impression that something wrong and unexpected happened, however many times events are intentionally "discarded" or "dropped" (we use the term "dropped" quite often, for example when we say that BeforeSend can drop events).

Reply:
I took the naming from Armins PR :P